app/features/send/spark-send-quote-hooks.ts

```diff
@@ -157,15 +158,25 @@ export function useOnSparkSendStateChange({
   const onCompletedRef = useLatest(onCompleted);
   const onFailedRef = useLatest(onFailed);
 
+  // Track the last triggered state for each quote to avoid duplicate callbacks
+  const lastTriggeredStateRef = useRef<Map<string, SparkSendQuote['state']>>(
```

Comment:
I decided to do this as well to make things a bit more efficient. Before this ever time the interval runs we would trigger the callback for the current state which was unnecessary. We only need it to run initially and then when the state changes.